src/v/redpanda/admin/api-doc/debug.json

```diff
@@ -325,9 +325,53 @@
                     ]
                 }
             ]
+        },
+        {
+            "path": "/v1/debug/cpu_profile",
+            "operations": [
+                {
+                    "method": "GET",
+                    "summary": "Gets the samples from the CPU profiler",
+                    "nickname": "cpu_profile",
+                    "produces": [
+                        "application/json"
+                    ],
+                    "type": "array",
+                    "items": {
+                        "type": "cpu_profile_sample"
+                    },
+                    "parameters": [
+                        {
+                            "name": "shard",
+                            "in": "query",
+                            "required": false,
+                            "allowMultiple": false,
+                            "type": "long"
+                        }
+                    ]
+                }
+            ]
         }
     ],
     "models": {
+        "cpu_profile_sample": {
+            "id": "cpu_profile_sample",
```

Comment:
You take a more fine grained approach here instead of just stringifying almost everything. We shortly talked about it the other day, do you think I should do the same now as well? I'd think probably yes if you do it here.

Reply:
We should discuss this in one of our stand ups. Obviously the return formats are not going to be identical since the information we want to return isn't. Specifically you'll have the allocation size data that isn't relevant to a CPU profiler. However, for data that overlaps I think we should make the return formats for both the admin API as well as the API barrier between seastar and RP as similar as possible.